### ISSUE TITLE
createAnimatedComponent should support stateless functional components [React Hooks]

### ISSUE THREAD

**Issue by IljaDaderko** [facebook/react-native#22066]
# For Discussion

I am starting to play with hooks in react native following guide posted by **hramos**
https://github.com/facebook/react-native/issues/21967#issuecomment-434113687

Off the bat I got an error

> `createAnimatedComponent` does not support stateless functional components; use a class instead.

This was as a result of me converting my simplest components (that don't use state) to functions and I assume will be an issue further down the road since vision of react for the future is to use less classes and more function components.

**Comment by MrLoh** [comment]
@hramos this is still an issue in 0.59.1, can this be reopened?

**Comment by adkenyon** [comment]
@MrLoh It looks as though createAnimatedComponent requires a ref (which functional components don't provide). You can get around the issue by wrapping using [React.forwardRef](https://reactjs.org/docs/forwarding-refs.html)

**Comment by MrLoh** [comment]
yeah, that was an obvious solution. Should have thought of that myself

**Comment by shubhamdeol** [comment]
@MrLoh  Can you please explain how to did this, using forwardRef ?

**Comment by MrLoh** [comment]
@shubhamdeol `const MyFunctionalComponent = React.forwardRef((props, ref) => <View ref={ref}> ... );` then you can do `createAnimatedComponent(MyFunctionalComponent);`

**Comment by shubhamdeol** [comment]
Actually. The component which is stateless is inside library. Can I pass ref to that without forking the library code?

**Comment by MrLoh** [comment]
Does the library component accept a `ref` prop? Not sure wether things work if Animated doesn't have a ref, you might have to ask the library maintainers to add a `ref` prop.

**Comment by shubhamdeol** [comment]
Okay, Thanks
I will do this.

**Comment by tberman** [comment]
I just tried this w/ a forwardRef and received the same error, and looking at the code of how forwardRef works, it is unclear it would pass this check. Does the forwardRef trick still work for people running on 16.8?
